flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/ConnectorCatalogTable.java

```diff
@@ -102,6 +102,11 @@ public boolean isBatch() {
 		throw new UnsupportedOperationException("ConnectorCatalogTable cannot be converted to properties");
 	}
 
+	@Override
+	public CatalogTable copy(Map<String, String> options) {
+		throw new UnsupportedOperationException("ConnectorCatalogTable cannot copy with new table options");
```

Comment:
Should we also support the path of `TableEnvironment.connect` to specify table option?

Reply:
All the table translating through `CatalogTableImpl` can have valid dynamic table options, including the table from `TableEnvironment.connect`. There is no way for a `ConnectorCatalogTable` to override table options because the table source is already there.